### DISCUSSION TITLE
App stuck on launchscreen on Xcode

### DISCUSSION THREAD

**Discussion by CelexDesign** [appcelerator/titanium_mobile#14010] [Q&A] [answered]
I created my first test app with Titanium and viewing it from the emulator launched by Visual Studio I see it correctly.

But after passing it to xCode the app no ​​longer launches on the emulator and remains stuck on the launch screen.

I use Xcode 15.2, VS Code ad Titatinum SDK 12.3.0.GA
![Screenshot 2024-03-19 alle 16 15 44](https://github.com/tidev/titanium-sdk/assets/5747801/a3b9ea3c-1f78-4c51-ab16-ce23baec23cf)

Thanks in advance for your help.

**Comment by m1ga** [comment, marked as answer]
hi, are you trying to build it with xcode? Can you try to run `ti build -p ios -T foo` in your project folder and then select "simulator" in the menu.

**Comment by CelexDesign** [comment]
Hi Michael, first of all thanks for your answer.

If I try the command "ti build -p ios -T foo" or if build the app with the standard VS procedure the app start correctly.
But, if I launch the App on xCode Emulator ​​its remains stuck on the launch screen. Is normally?

I would like to make sure that any switch from xCode (sign for Appstore or Enterprise) works correctly.
Or can I manage all finalizations directly from Visual Studio Code?

Very Thanks.

**Reply by m1ga** [reply]
I never use Xcode to do any compiling, I do all parts from the console (my workflow :smile: UI should work too). You can select your keys if you do a store/dist build from the menu. There is this PR https://github.com/tidev/titanium-sdk/pull/13032 that should add some functionality for standalone builds and I think there was a tutorial somewhere that gave some insights but I can't find it at the moment

**Reply by CelexDesign** [reply]
Hi Michael, 
if I can finalize and publish the app on the store directly from VS is fantastic for me 😄😄😄.

I'll try an app tomorrow.

Very, very, thanks.

**Reply by m1ga** [reply]
it will open xcode with the final archive at the end and from there you can do the normal upload process but you don't have to build it using xcode.
You do the steps starting from "3. Archive and upload your app using Xcode" in https://www.instabug.com/blog/how-to-submit-app-to-app-store the rest is done by the Ti build process.
There is a "package" button somewhere in VSCode:
![Screenshot_20240320_101139](https://github.com/tidev/titanium-sdk/assets/4334997/a3f0c886-282c-4196-9731-aee784389aac)


As mentioned before: I use the console as I find it easier (so there is always a fallback) and also using Pulsar as my editor and the build menu looks a bit different there:
![Screenshot_20240320_101055](https://github.com/tidev/titanium-sdk/assets/4334997/564d2561-85d4-40ec-b901-b15468360651) :smile: but most of the devs use VSCode so that should work
